src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Tokens/SecurityTokenHandlerCollectionManager.cs

```diff
@@ -0,0 +1,155 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using CoreWCF.IdentityModel.Configuration;
+
+namespace CoreWCF.IdentityModel.Tokens
+{
+    /// <summary>
+    /// A class which manages multiple named <see cref="SecurityTokenHandlerCollection"/>.
+    /// </summary>
+    public class SecurityTokenHandlerCollectionManager
+    {
+        private Dictionary<string, SecurityTokenHandlerCollection> _collections = new Dictionary<string, SecurityTokenHandlerCollection>();
+        private string _serviceName = ConfigurationStrings.DefaultServiceName;
+
```

Comment:
This gets set in both constructors so if you remove the field initialization here, ServiceName can be made into an auto property.